service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -218,8 +220,20 @@ public void inviteUser(JWTData userData, InviteUsersRequest inviteUserRequest, i
   }
 
   @Override
-  public UsersResponse getApplicants(int teamId) {
-    return getUsers(teamId, TeamRole.PENDING);
+  public UsersResponse getApplicants(JWTData userData, int teamId) {
+    checkTeamExists(teamId);
+    UsersTeamsRecord userTeamsRecord =
+            db.selectFrom(USERS_TEAMS)
+                    .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+                    .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+                    .and(USERS_TEAMS.TEAM_ROLE.eq(TeamRole.LEADER))
+                    .fetchOne();
```

Comment:
❓ This is a check that the user is team leader, right? Can this logic also be used for `disbandTeam`?

Reply:
Yes that's true, just abstracted that logic out because it's used all over the place.